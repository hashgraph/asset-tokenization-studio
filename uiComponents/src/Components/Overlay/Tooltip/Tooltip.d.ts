/// <reference types="react" />
import type { BaseSingleStyleConfiguration } from "@/Theme/Components/BaseSingleStyleConfiguration";
import type { TooltipProps as ChakraTooltipProps } from "@chakra-ui/tooltip";
export interface TooltipThemeConfiguration extends BaseSingleStyleConfiguration {
}
export interface TooltipProps extends ChakraTooltipProps {
}
export declare const Tooltip: ({ label, children, hasArrow, ...props }: TooltipProps) => JSX.Element;
