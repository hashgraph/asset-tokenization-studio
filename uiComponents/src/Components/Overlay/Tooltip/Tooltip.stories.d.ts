declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export default meta;
export declare const HoverText: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const HoverIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const CenterTop: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const LeftTop: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const RightTop: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const CenterBottom: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const LeftBottom: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const RightBottom: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const Left: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
export declare const Right: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Tooltip").TooltipProps>;
