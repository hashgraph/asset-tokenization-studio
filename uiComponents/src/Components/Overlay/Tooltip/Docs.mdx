import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as TooltipStories from "./Tooltip.stories";

<Meta of={TooltipStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1643%3A12710&mode=design&t=k4jSCm0Q182mRyVn-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

Add the following configurations to the `theme.tsx` and `Tooltip.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Tooltip: ConfigTooltip,
  },
};
```

```tsx dark
const ConfigTooltip = {
  baseStyle: {},
  variants: {},
  defaultProps: {},
};
```

## Props

<ArgTypes of={TooltipStories} />

## Stories

<Stories title="" />
