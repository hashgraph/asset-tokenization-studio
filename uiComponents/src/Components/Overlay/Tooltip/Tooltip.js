var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { Flex as ChrakraFlex, Tooltip as ChakraTooltip, } from "@chakra-ui/react";
export var Tooltip = function (_a) {
    var label = _a.label, children = _a.children, _b = _a.hasArrow, hasArrow = _b === void 0 ? true : _b, props = __rest(_a, ["label", "children", "hasArrow"]);
    return (React.createElement(ChakraTooltip, __assign({ "data-testid": "tooltip", label: label, hasArrow: hasArrow }, props),
        React.createElement(ChrakraFlex, null, children)));
};
