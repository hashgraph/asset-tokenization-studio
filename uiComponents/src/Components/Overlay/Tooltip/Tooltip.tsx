import React from "react";
import type { BaseSingleStyleConfiguration } from "@/Theme/Components/BaseSingleStyleConfiguration";
import {
  Flex as ChrakraFlex,
  Tooltip as ChakraTooltip,
} from "@chakra-ui/react";
import type { TooltipProps as ChakraTooltipProps } from "@chakra-ui/tooltip";

export interface TooltipThemeConfiguration
  extends BaseSingleStyleConfiguration {}

export interface TooltipProps extends ChakraTooltipProps {}

export const Tooltip = ({
  label,
  children,
  hasArrow = true,
  ...props
}: TooltipProps) => {
  return (
    <ChakraTooltip
      data-testid="tooltip"
      label={label}
      hasArrow={hasArrow}
      {...props}
    >
      <ChrakraFlex>{children}</ChrakraFlex>
    </ChakraTooltip>
  );
};
