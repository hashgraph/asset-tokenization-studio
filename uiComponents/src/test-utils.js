import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
