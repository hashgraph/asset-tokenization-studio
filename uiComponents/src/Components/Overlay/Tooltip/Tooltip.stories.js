var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { Button, Flex } from "@chakra-ui/react";
import { Tooltip } from "./Tooltip";
import { Info } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
var meta = {
    title: "Design System/Overlay/Tooltip",
    component: Tooltip,
    argTypes: {
        placement: {
            description: "Tooltip placement",
            options: [
                "auto-start",
                "auto",
                "auto-end",
                "top-start",
                "top",
                "top-end",
                "right-start",
                "right",
                "right-end",
                "bottom-start",
                "bottom",
                "bottom-end",
                "left-start",
                "left",
                "left-end",
            ],
            control: {
                type: "select"
            }
        },
        hasArrow: {
            description: "Tooltip with arrow",
            options: [true, false],
            control: {
                type: "select"
            }
        },
        label: {
            description: "Tooltip label",
            defaultValue: "tooltip",
            control: {
                type: "object"
            }
        },
        variant: {
            description: "Variant",
            options: ["light", "dark"],
            control: {
                type: "select"
            }
        },
        children: {
            control: {
                type: null
            }
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A6678"
        }
    },
    args: {
        trigger: "hover",
        children: React.createElement(Button, null, "hover me!"),
        label: "Here’s a regular tooltip with some text inside of it that’s supposed to be substantially large."
    }
};
export default meta;
var Template = function (_a) {
    var label = _a.label, args = __rest(_a, ["label"]);
    return (React.createElement(Flex, { align: "center", pos: "relative", maxW: "10px" },
        React.createElement(Tooltip, __assign({ label: label }, args))));
};
export var HoverText = Template.bind({});
HoverText.args = {
    placement: "top"
};
export var HoverIcon = Template.bind({});
HoverIcon.args = {
    children: React.createElement(PhosphorIcon, { as: Info, size: "small", p: "2px" })
};
export var CenterTop = Template.bind({});
CenterTop.args = {
    placement: "top",
    label: "Here's a short tooltip"
};
export var LeftTop = Template.bind({});
LeftTop.args = {
    placement: "top-start",
    label: "Here's a short tooltip"
};
export var RightTop = Template.bind({});
RightTop.args = {
    placement: "top-end",
    label: "Here's a short tooltip"
};
export var CenterBottom = Template.bind({});
CenterBottom.args = {
    placement: "bottom",
    label: "Here's a short tooltip"
};
export var LeftBottom = Template.bind({});
LeftBottom.args = {
    placement: "bottom-start",
    label: "Here's a short tooltip"
};
export var RightBottom = Template.bind({});
RightBottom.args = {
    placement: "bottom-end",
    label: "Here's a short tooltip"
};
export var Left = Template.bind({});
Left.args = {
    placement: "left",
    label: "Here's a short tooltip"
};
export var Right = Template.bind({});
Right.args = {
    placement: "right",
    label: "Here's a short tooltip"
};
