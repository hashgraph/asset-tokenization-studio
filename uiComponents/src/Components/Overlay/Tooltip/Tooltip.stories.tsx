import React from "react";
import { Button, Flex } from "@chakra-ui/react";
import type { Meta, StoryFn } from "@storybook/react";

import { Tooltip } from "./Tooltip";
import { Info } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";

const meta = {
  title: "Design System/Overlay/Tooltip",
  component: Tooltip,
  argTypes: {
    placement: {
      description: "Tooltip placement",
      options: [
        "auto-start",
        "auto",
        "auto-end",
        "top-start",
        "top",
        "top-end",
        "right-start",
        "right",
        "right-end",
        "bottom-start",
        "bottom",
        "bottom-end",
        "left-start",
        "left",
        "left-end",
      ],
      control: {
        type: "select",
      },
    },
    hasArrow: {
      description: "Tooltip with arrow",
      options: [true, false],
      control: {
        type: "select",
      },
    },
    label: {
      description: "Tooltip label",
      defaultValue: "tooltip",
      control: {
        type: "object",
      },
    },
    variant: {
      description: "Variant",
      options: ["light", "dark"],
      control: {
        type: "select",
      },
    },
    children: {
      control: {
        type: null,
      },
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A6678",
    },
  },
  args: {
    trigger: "hover",
    children: <Button>hover me!</Button>,
    label:
      "Here’s a regular tooltip with some text inside of it that’s supposed to be substantially large.",
  },
} as Meta<typeof Tooltip>;
export default meta;

const Template: StoryFn<typeof Tooltip> = ({ label, ...args }) => (
  <Flex align="center" pos="relative" maxW="10px">
    <Tooltip label={label} {...args}></Tooltip>
  </Flex>
);

export const HoverText = Template.bind({});
HoverText.args = {
  placement: "top",
};

export const HoverIcon = Template.bind({});
HoverIcon.args = {
  children: <PhosphorIcon as={Info} size="small" p="2px" />,
};

export const CenterTop = Template.bind({});
CenterTop.args = {
  placement: "top",
  label: "Here's a short tooltip",
};

export const LeftTop = Template.bind({});
LeftTop.args = {
  placement: "top-start",
  label: "Here's a short tooltip",
};

export const RightTop = Template.bind({});
RightTop.args = {
  placement: "top-end",
  label: "Here's a short tooltip",
};

export const CenterBottom = Template.bind({});
CenterBottom.args = {
  placement: "bottom",
  label: "Here's a short tooltip",
};

export const LeftBottom = Template.bind({});
LeftBottom.args = {
  placement: "bottom-start",
  label: "Here's a short tooltip",
};

export const RightBottom = Template.bind({});
RightBottom.args = {
  placement: "bottom-end",
  label: "Here's a short tooltip",
};

export const Left = Template.bind({});
Left.args = {
  placement: "left",
  label: "Here's a short tooltip",
};

export const Right = Template.bind({});
Right.args = {
  placement: "right",
  label: "Here's a short tooltip",
};
