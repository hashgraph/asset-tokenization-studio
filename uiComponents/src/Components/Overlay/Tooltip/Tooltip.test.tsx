import { render } from "@/test-utils";
import { waitFor, fireEvent, screen } from "@testing-library/react";
import React from "react";
import type { TooltipProps } from "./Tooltip";
import { Tooltip } from "./Tooltip";

const HOVER = "Hover";
const TEST = "Test";

describe(`<Tooltip />`, () => {
  const DummyComponent = (
    props: Omit<TooltipProps & { isButtonDisabled?: boolean }, "children">
  ) => {
    const { isButtonDisabled, ...tooltipProps } = props;
    return (
      <Tooltip label={TEST} {...tooltipProps}>
        <button disabled={isButtonDisabled || false}>{HOVER}</button>
      </Tooltip>
    );
  };
  const factoryComponent = (props?: TooltipProps) =>
    render(<DummyComponent {...props} />);

  test("renders correctly default", async () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
    expect(component.getByText(HOVER)).toBeInTheDocument();

    fireEvent.pointerOver(screen.getByText(HOVER));

    await waitFor(() => {
      expect(component.getByText(TEST)).toBeInTheDocument();
    });
  });
});
