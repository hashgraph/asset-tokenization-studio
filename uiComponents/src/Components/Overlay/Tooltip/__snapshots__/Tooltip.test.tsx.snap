// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders correctly default 1`] = `
<DocumentFragment>
  <div
    class="css-k008qs"
  >
    <button>
      Hover
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
